### PR TITLE
Properly handle errors on void methods

### LINKED ISSUE #574: EmptyBodyDeserializer does not handle error responses
## What happened?
A void endpoint that returned an error did not result in an exception. The `EmptyBodyDeserializer` does not decode errors:
https://github.com/palantir/dialogue/blob/bb92eac336ba9e2e0938ed1d6b4248f61114fee9/dialogue-serde/src/main/java/com/palantir/conjure/java/dialogue/serde/ConjureBodySerDe.java#L280-L284

### PR DESCRIPTION
**Pull request by ferozco**
Closes #574 
## Before this PR
We would silently drop errors on endpoints that did not expect a response

## After this PR
==COMMIT_MSG==
Properly handle errors on void methods
==COMMIT_MSG==

## Possible downsides?
N/A

### CODE REVIEW

File: dialogue-serde/src/main/java/com/palantir/conjure/java/dialogue/serde/ConjureBodySerDe.java [hunk 1]
```diff
@@ -272,15 +274,25 @@ public String toString() {
...
-            response.close();
-            return null;
+            try {
+                if (errorDecoder.isError(response)) {
+                    throw errorDecoder.decode(response);
+                }
+                return null;
+            } finally {
+                // We should not fail if a server that previously returned nothing starts returning a response
+                response.close();
```
Comment (iamdanfox) [inline dialogue-serde/src/main/java/com/palantir/conjure/java/dialogue/serde/ConjureBodySerDe.java]: could use try with resources here if you like

File: dialogue-serde/src/test/java/com/palantir/conjure/java/dialogue/serde/ConjureBodySerDeTest.java [hunk 2]
```diff
@@ -196,6 +196,32 @@ public void testBinary_optional_empty() {
...
+        response.code = 204;
+        BodySerDe serializers =
+                new ConjureBodySerDe(ImmutableList.of(WeightedEncoding.of(new StubEncoding("application/json"))));
+        serializers.emptyBodyDeserializer().deserialize(response);
+    }
+
+    @Test
+    public void testEmptyResponse_failure() {
+        TestResponse response = new TestResponse();
+        response.code = 400;
```
Comment (iamdanfox) [inline dialogue-serde/src/test/java/com/palantir/conjure/java/dialogue/serde/ConjureBodySerDeTest.java]: if you merge in develop, I've just changed this to be a setter method, so `new TestResponse().code(400)`

**Review by iamdanfox — APPROVED** [review]

**Comment by ferozco** [comment]
Pending https://github.com/palantir/conjure-verification/pull/326

**Review by carterkozak — APPROVED** [review]
updated. I think there's an issue with the `receiveBinaryAliasExample` confirmation, not necessarily a client issue, so I've added the two cases to `ignored-test-cases.yml`.

**Comment by ferozco** [comment]
Thanks @carterkozak!

**Pull request merged**